hierarchicalforecast/core.py

```diff
@@ -154,7 +155,7 @@ def _prepare_fit(self,
         model_names = [name for name in model_names if name not in pi_model_names]
         
         # TODO: Complete y_hat_insample protection
-        if intervals_method in ['bootstrap', 'permbu']:
+        if intervals_method in ['bootstrap', 'permbu'] and Y_df is not None:
```

Comment:
Good catch!

Reply:
Technically it isn't necessary because we catch it earlier but mypy won't shutup if I don't add it here.....